packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/rest/networkAnalyticsClient.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import { logger } from "../logger.js";
+import { TokenCredential } from "@azure/core-auth";
+import { NetworkAnalyticsContext } from "./clientDefinitions.js";
+
+/**
+ * Initialize a new instance of `NetworkAnalyticsContext`
+ * @param credentials - uniquely identify client credential
+ * @param options - the parameter for all optional parameters
+ */
+export default function createClient(
+  credentials: TokenCredential,
+  options: ClientOptions = {},
+): NetworkAnalyticsContext {
+  const endpointUrl =
+    options.endpoint ?? options.baseUrl ?? `https://management.azure.com`;
```

Comment:
this issue is fixed in latest generation https://github.com/Azure/autorest.typescript/issues/2349